src/NuGet.Services.Metadata.Catalog.Monitoring/Validation/Test/Registration/RegistrationDeprecationValidator.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Logging;
+using NuGet.Services.Metadata.Catalog.Monitoring.Model;
+using NuGet.Services.Metadata.Catalog.Monitoring.Validation.Test.Exceptions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring.Validation.Test.Registration
+{
+    public class RegistrationDeprecationValidator : RegistrationIndexValidator
+    {
+        public RegistrationDeprecationValidator(
+            ValidatorConfiguration config,
+            ILogger<RegistrationDeprecationValidator> logger)
+            : base(config, logger)
+        {
+        }
+
+        public override Task CompareIndexAsync(ValidationContext context, PackageRegistrationIndexMetadata database, PackageRegistrationIndexMetadata v3)
+        {
+            var exceptions = new List<MetadataFieldInconsistencyException<PackageRegistrationIndexMetadata>>();
+
+            if (database.Deprecation == null || v3.Deprecation == null)
+            {
+                if (database.Deprecation != null || v3.Deprecation != null)
+                {
+                    throw new MetadataFieldInconsistencyException<PackageRegistrationIndexMetadata>(
+                        database, v3,
+                        nameof(PackageRegistrationIndexMetadata.Deprecation),
+                        i => i.Deprecation);
+                }
+
+                return Task.CompletedTask;
+            }
+
+            if (!database.Deprecation.Reasons.SequenceEqual(v3.Deprecation.Reasons))
+            {
+                AddDeprecationInconsistencyException(
+                    exceptions,
+                    database, v3,
+                    nameof(PackageRegistrationDeprecationMetadata.Reasons),
+                    d => d.Reasons);
+            }
+
+            if (database.Deprecation.Message != v3.Deprecation.Message)
+            {
+                AddDeprecationInconsistencyException(
+                    exceptions,
+                    database, v3,
+                    nameof(PackageRegistrationDeprecationMetadata.Message),
+                    d => d.Message);
+            }
+
+            if (database.Deprecation.AlternatePackage == null || v3.Deprecation.AlternatePackage == null)
```

Comment:
Similar readability issue here. Although technically correct, I think it could be made more readable by simplifying the logical constructs.

```
if (database.Deprecation.AlternatePackage != null && v3.Deprecation.AlternatePackage != null)
{
  if (database.Deprecation.AlternatePackage.Id != v3.Deprecation.AlternatePackage.Id)
  {
    AddAlternatePackageInconsistencyException(
                        exceptions,
                        database, v3,
                        nameof(PackageRegistrationAlternatePackageMetadata.Id),
                        a => a.Id);
  }

  if (database.Deprecation.AlternatePackage.Range != v3.Deprecation.AlternatePackage.Range)
  {
    AddAlternatePackageInconsistencyException(
                        exceptions,
                        database, v3,
                        nameof(PackageRegistrationAlternatePackageMetadata.Range),
                        a => a.Range);
  }
}
else if ((database.Deprecation.AlternatePackage == null && v3.Deprecation.AlternatePackage != null)
  || (database.Deprecation.AlternatePackage != null && v3.Deprecation.AlternatePackage == null))
{
  AddDeprecationInconsistencyException(
                        exceptions,
                        database, v3,
                        nameof(PackageRegistrationDeprecationMetadata.AlternatePackage),
                        d => d.AlternatePackage);
}
```

Reply:
I moved the alternate package verifications to their own function and then used the structure you suggested above